packages/thirdweb/src/react/web/ui/components/WalletImage.tsx

```diff
@@ -23,18 +26,33 @@ export function WalletImage(props: {
   id: WalletId;
   size: string;
   client: ThirdwebClient;
-  allowOverrides?: boolean;
 }) {
   const [image, setImage] = useState<string | undefined>(undefined);
+  const activeWallet = useActiveWallet();
   useEffect(() => {
     async function fetchImage() {
+      // show EOA icon for external wallets
+      // show auth provider icon for in-app wallets
+      // show the admin EOA icon for smart
+      const storage = getStorage();
+      let activeEOAId = props.id;
+      if (props.id === "smart") {
+        const storedId = await getStoredActiveWalletId(storage);
```

Comment:
Instead of relying on storage - we can add a new state for storing personal wallet in the wallet manager - this will also be needed to implement "Switch to personal wallet" button

Reply:
hm yeah that's better - would i add a new 'store' for personalWallet and read/write from that

Reply:
actually i dont think we need yet another store, we already have the connectWallets store and we know the personal wallet id so we can already implement a switch to personal wallet that way